### DISCUSSION TITLE
Rest API Call for create release always return 422 invalid tag.

### DISCUSSION THREAD

**Discussion by mai1x9** [community/community#32303] [Apps, API and Webhooks] [answered]
Hi,
I am trying to create release using rest api programatically using python.
I have tried even using curl, but it resulted in 422, invalid tag.

I have referenced to the rest api document by github at https://docs.github.com/en/rest/releases/releases#create-a-release\

The curl script which i have tried 
```
curl -X POST -H "Accept: application/vnd.github+json" -H "Authorization: Bearer XXXXXXXXXXXXXXX" "https://api.github.com/repos/mai1x9/releasetest/releases" -d '{"tag_name":"v2.0.0","target_commitish":"master","name":"v2.0.0","body":"Description of the release","draft":false,"prerelease":false,"generate_release_notes":false}'
```

and the python script that was used,
```
import requests
import json

target_url = "https://api.github.com/repos/mai1x9/releasetest/releases"
release_json = {
    "owner":"mai1x9",
    "repo": "releasetest",
    "tag_name": 'v1.0',
    "target_commitish": 'master',
    "name": 'v1.0',
    "body": 'Description of the release',
    "draft": False,
    "prerelease": False,
    "generate_release_notes": False

}

gh_headers = {
    "Accept": "application/vnd.github+json",
    "Authorization": "Bearer XXXXXXXXXX"
}

r = requests.post(url=target_url, data=json.dumps(release_json), headers=gh_headers)
print(r.status_code)
print(r.json())
```

**In either cases, the response is,**
```
{
  "message": "Validation Failed",
  "errors": [
    {
      "resource": "Release",
      "code": "custom",
      "field": "tag_name",
      "message": "tag_name is not a valid tag"
    },
    {
      "resource": "Release",
      "code": "custom",
      "message": "Published releases must have a valid tag"
    },
    {
      "resource": "Release",
      "code": "invalid",
      "field": "target_commitish"
    }
  ],
  "documentation_url": "https://docs.github.com/rest/reference/repos#create-a-release"
}

```

I have passed Personal Access Token under header for authorization. 

Sorry for posting the question here under discussions, but I found no where, where to raise the issues related to github rest api.

Any help would be really great and would be thankful.

Thank you.

On Side Note,
- List release API is working as expected using python and curl, where bearer token for authroization is PAT.
- I have given full access to the PAT token.

**Comment by mai1x9** [comment, marked as answer]
My bad, github renamed the default branch from master to main, which got skipped out of my mind. Due to invalid branch name, it throwed the error.

**Reply by HomeOfTheWizard** [reply]
I had exactly the same issue. Thanks to you I could resolve. but IMHO the error message should be improved. It is difficult to understand the real issue with that message.
